src/Admin/AdminHelper.php

```diff
@@ -338,19 +338,36 @@ public function getElementAccessPath($elementId, $entity)
 
     /**
      * Recursively find the class name of the admin responsible for the element at the end of an association chain.
-     *
-     * @param array $elements
-     *
-     * @return string
      */
-    protected function getEntityClassName(AdminInterface $admin, $elements)
+    protected function getModelClassName(AdminInterface $admin, array $elements): string
```

Comment:
Maybe we should make this method final, for the reason stated above?
```suggestion
    final protected function getModelClassName(AdminInterface $admin, array $elements): string
```

Reply:
I've restored these calls in order to respect the BC promise.